core/validation/pre_transformation_validate.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+import pandas as pd
+
+import core.validation.failures.user as uf
+import core.validation.messages as msgs
+from core.const import PLACE_TO_FUND_TYPE, PreTransformationCheck
+from core.exceptions import ValidationError
+from core.validation.pre_transformation_validation_schema import TF_ROUND_3, TF_ROUND_4
+
+
+def pre_transformation_validations(
+    workbook: dict[str, pd.DataFrame],
+    reporting_round: int,
+    place_names: tuple,
+    fund_types: tuple,
+):
+    """Performs pre-transformation validations based on the provided schema.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the submission.
+    :param reporting_round: The reporting round for which the validations are performed.
+    :param place_names: A tuple of valid place names.
+    :param fund_types: A tuple of valid fund types.
+    :raises ValidationError: If any of the validation functions catch a validation error.
+    """
+    if reporting_round in [1, 2]:
+        return  # do not do pre-transformation validation on historical data
+
+    round_to_schema = {3: TF_ROUND_3, 4: TF_ROUND_4}
+
+    schema = round_to_schema[reporting_round]
+
+    wrong_input_validation(workbook, schema)  # first, check if valid
+    conflicting_input_validation(workbook, schema)
+
+    if reporting_round == 4:
+        authorisation_validation(workbook, place_names, fund_types, schema)
+        # check last as do not want raise authorisation error if input is invalid or conflicting
+
+
+def authorisation_validation(workbook: dict[str, pd.DataFrame], place_names: tuple, fund_types: tuple, schema: dict):
+    """Performs authorisation validation based on the provided schema.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                    DataFrames representing each sheet in the submission.
+    :param place_names: A tuple of valid place names.
+    :param fund_types: A tuple of valid fund types.
+    :param schema: The schema containing authorisation checks.
+    :raises ValidationError: If authorisation validation fails.
+    """
+    type_to_authorised_values = {
+        "Place Name": place_names,
+        "Fund Type": fund_types,
+    }
+    authorisation_checks = list()
+    for sheet, column, row, expected_values, type in schema["authorisation_checks"]:
+        expected_values = type_to_authorised_values[type]
+
+        authorisation_checks.append(
+            PreTransformationCheck(sheet=sheet, column=column, row=row, expected_values=expected_values, type=type)
+        )
+
+    failures = check_values(workbook, authorisation_checks, "unauthorised_submission_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
```

Comment:
Yeh, that makes a lot of sense. In my head I was thinking that place_name and fund_type would be universal variables across funds, but that won't necessarily be the case. This change may well lead to some downtime on prod though. 

If we change the authorisation in data-store to expect a JSON object, then we also have to change Submit's payload structure, but it might be a bit of a chicken-and-the-egg problem in that a change on one repo by itself will cause the other to fall over. There's ways around this but it will require some forethought.

Reply:
I would argue a bit of planned downtime is ok. We can do it at a time of low traffic and the impact will be minimal to non-existent.

Reply:
I raise the issue because @gidsg asked me to write the changes such that they could be swapped in and out without any downtime on production.

Something like this can potentially achieve that:

```
auth = body.get("auth", {"Places Names": body.get("place_names"),
                             "Fund Types": {"Town_Deal", "Future_High_Street_Fund"}})
```

Bit hacky, but then we swap out to ```auth = body.get("auth")``` once the Submit changes are also in. There's also the problem of the private contractor mappings.

If we could get 10 minutes of downtime we can make the switch without this, but what's your opinion @gidsg?